### ISSUE TITLE
On Windows IdentitiesOnly with public identity file through SSH agent not authenticating

### ISSUE THREAD

**Issue by sferrazzi** [iterate-ch/cyberduck#14196]
SFTP access.
It works from the Windows shell, but I get an error with Cyberduck:

```
Host hhh
  HostName hhh.cloudapp.azure.com
  User hhh-admin
  IdentityFile \Users\hhh\.ssh\hhh-admin-key.pub
  IdentitiesOnly yes
```

```
ch.cyberduck.core.AbstractExceptionMappingService - No message for failure net.schmizz.sshj.common.SSHException
ch.cyberduck.core.threading.DefaultFailureDiagnostics - Determine cause for failure BackgroundException{class=class ch.cyberduck.core.exception.BackgroundException, file=null, message='Accesso fallito.', detail='Sconosciuto.', cause='net.schmizz.sshj.common.SSHException'}
```

**Comment by dkocher** [comment]
Try to use a path using UNIX path delimiters when referencing the public key as in 

```
IdentityFile ~/.ssh/hhh-admin-key.pub
```

**Comment by sferrazzi** [comment]
> Try to use a path using UNIX path delimiters when referencing the public key as in
> 
> ```
> IdentityFile ~/.ssh/hhh-admin-key.pub
> ```

Same problem.

**Comment by sferrazzi** [comment]
1Password and Windows 11 system.

**Comment by sferrazzi** [comment]
With IdentityFile ~/.ssh/hhh-admin-key.pub:

WARN  ch.cyberduck.core.sftp.auth.SFTPAgentAuthentication - Missing IdentityFile configuration for Host{protocol=Profile{parent=sftp, vendor=iterate GmbH, description=null, image=null}, region='null', port=22, hostname='hhh', credentials=Credentials{user='hhh-admin', oauth='OAuthTokens{accessToken='', refreshToken='', expiryInMilliseconds=9223372036854775807}', token='', identity=Local{path='~\.ssh\hhh-admin-key.pub'}}, uuid='48aaf26a-0007-4332-a98c-3dff9e9964a1', nickname='hhh', defaultpath='null', workdir=null, custom=null, labels=null}

**Comment by ylangisc** [comment]
It probably should be `/Users/hhh/.ssh/hhh-admin-key.pub` as `~` is not expanded on Windows.

**Comment by AliveDevil** [comment]
I just configured 1Password with a public key file, saved the public key in `%userprofile%\.ssh`, and configured the ssh config file as

```
Host hostname
  IdentityFile C:\Users\user\.ssh\host_ed25519.pub
  IdentitiesOnly yes
```

And this works just fine. The private key isn't saved to disk, but access through the 1Password agent.

**Comment by AliveDevil** [comment]
Reprod.

[cyberduck.log](https://github.com/iterate-ch/cyberduck/files/10661875/cyberduck.log)

**Comment by sferrazzi** [comment]
> It probably should be `/Users/hhh/.ssh/hhh-admin-key.pub` as `~` is not expanded on Windows.

I tried, but creating new Bookmark the "SSH Private Key" is filled with the public key path: /Users/hhh/.ssh/hhh-admin-key.pub
I tried with "SSH Private Key" to none: same error.

**Comment by sferrazzi** [comment]
> Reprod.
> 
> ssh config:
> 
> ```
> Host github-14196.iterate.lab.xc
> 	IdentityFile C:\Users\alive\.ssh\14196_ed25519.pub
> 	IdentitiesOnly yes
> ```
> 
> [cyberduck.log](https://github.com/iterate-ch/cyberduck/files/10661875/cyberduck.log)

Ok, thank you.

**Comment by AliveDevil** [comment]
Not an issue with either 1Password or Cyberduck Core (macOS works fine), rather a generic issue on Windows with OpenSSH agent authentication (using IdentityFile to a public key, and IdentitiesOnly) - reproduced with ssh-agent.

**Comment by sferrazzi** [comment]
> Not an issue with either 1Password or Cyberduck Core (macOS works fine), rather a generic issue on Windows with OpenSSH agent authentication (using IdentityFile to a public key, and IdentitiesOnly) - reproduced with ssh-agent.

Windows shell;

sftp hhh@hhh

work correctly on my PC.

**Comment by AliveDevil** [comment]
We have an issue with our order-of-execution.

Following happens:
- Try authentication with PuTTy Pageant first[^pagent]
   Which fails (e.g. pageant not running)
- Try authentication with OpenSSH agent[^agent]
   This fails, as the server already closed the connection (I don't know why)

[^pagent]: https://github.com/iterate-ch/cyberduck/blob/master/ssh/src/main/java/ch/cyberduck/core/sftp/SFTPSession.java#L281
[^agent]: https://github.com/iterate-ch/cyberduck/blob/master/ssh/src/main/java/ch/cyberduck/core/sftp/SFTPSession.java#L283

1. Pagent fails with:
```
net.schmizz.sshj.userauth.UserAuthException
  HResult=0x80131500
  Nachricht = Exhausted available authentication methods
  Quelle = Cyberduck.Protocols
  Stapelüberwachung:
   bei net.schmizz.sshj.SSHClient.auth(String username, Iterable methods) in SSHClient.java: Zeile230
   bei net.schmizz.sshj.SSHClient.auth(String username, AuthMethod[] methods) in SSHClient.java: Zeile206
   bei ch.cyberduck.core.sftp.auth.SFTPAgentAuthentication.authenticate(Host bookmark, LoginCallback prompt, CancelCallback cancel) in SFTPAgentAuthentication.java: Zeile101

  Diese Ausnahme wurde ursprünglich von dieser Aufrufliste ausgelöst:
    net.schmizz.sshj.SSHClient.auth(string, java.lang.Iterable) in SSHClient.java
    net.schmizz.sshj.SSHClient.auth(string, net.schmizz.sshj.userauth.method.AuthMethod[]) in SSHClient.java
    ch.cyberduck.core.sftp.auth.SFTPAgentAuthentication.authenticate(ch.cyberduck.core.Host, ch.cyberduck.core.LoginCallback, ch.cyberduck.core.threading.CancelCallback) in SFTPAgentAuthentication.java
```

2. The sshd-service writes to Journal:
```
Connection closed by authenticating user ubuntu 192.168.178.119 port 64772 [preauth]
```

3. The OpenSSH agent then fails with
```
java.lang.IllegalStateException
  HResult=0x80131500
  Nachricht = Not connected
  Quelle = Cyberduck.Protocols
  Stapelüberwachung:
   bei net.schmizz.sshj.SSHClient.checkConnected() in SSHClient.java: Zeile849
   bei net.schmizz.sshj.SSHClient.auth(String username, AuthMethod[] methods) in SSHClient.java: Zeile205
   bei ch.cyberduck.core.sftp.auth.SFTPAgentAuthentication.authenticate(Host bookmark, LoginCallback prompt, CancelCallback cancel) in SFTPAgentAuthentication.java: Zeile101
   bei ch.cyberduck.core.sftp.auth.SFTPAgentAuthentication.<bridge>authenticate(Host h, LoginCallback lc, CancelCallback cc) in SFTPAgentAuthentication.java: Zeile58
   bei ch.cyberduck.core.sftp.auth.SFTPAgentAuthentication.__<overridestub>ch.cyberduck.core.AuthenticationProvider::authenticate(Host , LoginCallback , CancelCallback )
   bei ch.cyberduck.core.sftp.SFTPSession.authenticate(SSHClient client, Host host, LoginCallback prompt, CancelCallback cancel) in SFTPSession.java: Zeile346

  Diese Ausnahme wurde ursprünglich von dieser Aufrufliste ausgelöst:
    net.schmizz.sshj.SSHClient.checkConnected() in SSHClient.java
    net.schmizz.sshj.SSHClient.auth(string, net.schmizz.sshj.userauth.method.AuthMethod[]) in SSHClient.java
    ch.cyberduck.core.sftp.auth.SFTPAgentAuthentication.authenticate(ch.cyberduck.core.Host, ch.cyberduck.core.LoginCallback, ch.cyberduck.core.threading.CancelCallback) in SFTPAgentAuthentication.java
    ch.cyberduck.core.sftp.auth.SFTPAgentAuthentication.<bridge>authenticate(ch.cyberduck.core.Host, ch.cyberduck.core.LoginCallback, ch.cyberduck.core.threading.CancelCallback) in SFTPAgentAuthentication.java
    ch.cyberduck.core.sftp.auth.SFTPAgentAuthentication.ch.cyberduck.core.AuthenticationProvider.AuthenticationProvider::authenticate(ch.cyberduck.core.Host, ch.cyberduck.core.LoginCallback, ch.cyberduck.core.threading.CancelCallback)
    ch.cyberduck.core.sftp.SFTPSession.authenticate(net.schmizz.sshj.SSHClient, ch.cyberduck.core.Host, ch.cyberduck.core.LoginCallback, ch.cyberduck.core.threading.CancelCallback) in SFTPSession.java
```
